cmd/stupid/main.go

```diff
@@ -20,60 +21,97 @@ func main() {
 			logger.SetLevel(lvl)
 		}
 	}
-	if len(os.Args) != 3 {
-		fmt.Printf("Usage: stupid config.yaml 500\n")
+	err := process(logger)
+	if err != nil {
+		fmt.Println(err)
 		os.Exit(1)
 	}
-	config := infra.LoadConfig(os.Args[1])
+	os.Exit(0)
+}
+
+func process(logger *log.Logger) error {
+	if len(os.Args) != 3 {
+		return errors.Errorf("error input parameters for stupid: stupid config.yaml 500\n")
+	}
 	N, err := strconv.Atoi(os.Args[2])
 	if err != nil {
-		panic(err)
+		return errors.Errorf("error input parameters for stupid: stupid config.yaml 500\n")
+	}
+	config, err := infra.LoadConfig(os.Args[1], logger)
+	if err != nil {
+		return err
+	}
+	crypto, err := config.LoadCrypto(logger)
+	if err != nil {
+		return err
 	}
-	crypto := config.LoadCrypto()
-
 	raw := make(chan *infra.Elements, 100)
 	signed := make([]chan *infra.Elements, len(config.Endorsers))
 	processed := make(chan *infra.Elements, 10)
 	envs := make(chan *infra.Elements, 10)
 	done := make(chan struct{})
-
+	finishCh := make(chan int)
+	defer close(finishCh)
+	errorC := make(chan error, 10)
+	defer close(errorC)
```

Comment:
for naming consistency, you could append `C` or `Ch` to every var here (pls be consistent)